mediapipe/tasks/python/test/vision/image_classifier_test.py

```diff
@@ -0,0 +1,139 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for image classifier."""
+
+import enum
+
+import numpy as np
+from absl.testing import absltest
+from absl.testing import parameterized
+
+from mediapipe.python._framework_bindings import image as image_module
+from mediapipe.tasks.python.components.processors import classifier_options
+from mediapipe.tasks.python.components.containers import category as category_module
+from mediapipe.tasks.python.components.containers import classifications as classifications_module
+from mediapipe.tasks.python.core import base_options as base_options_module
+from mediapipe.tasks.python.test import test_utils
+from mediapipe.tasks.python.vision import image_classifier
+from mediapipe.tasks.python.vision.core import vision_task_running_mode as running_mode_module
+
+_BaseOptions = base_options_module.BaseOptions
+_ClassifierOptions = classifier_options.ClassifierOptions
+_Category = category_module.Category
+_ClassificationEntry = classifications_module.ClassificationEntry
+_Classifications = classifications_module.Classifications
+_ClassificationResult = classifications_module.ClassificationResult
+_Image = image_module.Image
+_ImageClassifier = image_classifier.ImageClassifier
+_ImageClassifierOptions = image_classifier.ImageClassifierOptions
+_RUNNING_MODE = running_mode_module.VisionTaskRunningMode
+
+_MODEL_FILE = 'mobilenet_v2_1.0_224.tflite'
+_IMAGE_FILE = 'burger.jpg'
+_EXPECTED_CLASSIFICATION_RESULT = _ClassificationResult(
+  classifications=[
+    _Classifications(
+      entries=[
+        _ClassificationEntry(
+          categories=[
+            _Category(
+              index=934,
+              score=0.7939587831497192,
+              display_name='',
+              category_name='cheeseburger'),
+            _Category(
+              index=932,
+              score=0.02739289402961731,
+              display_name='',
+              category_name='bagel'),
+            _Category(
+              index=925,
+              score=0.01934075355529785,
+              display_name='',
+              category_name='guacamole'),
+            _Category(
+              index=963,
+              score=0.006327860057353973,
+              display_name='',
+              category_name='meat loaf')
+          ],
+          timestamp_ms=0
+        )
+      ],
+      head_index=0,
+      head_name='probability')
+  ])
+_ALLOW_LIST = ['cheeseburger', 'guacamole']
+_DENY_LIST = ['cheeseburger']
+_SCORE_THRESHOLD = 0.5
+_MAX_RESULTS = 3
+
+
+class ModelFileType(enum.Enum):
+  FILE_CONTENT = 1
+  FILE_NAME = 2
+
+
+class ImageClassifierTest(parameterized.TestCase):
```

Comment:
Better to test more corner cases such as the invalid configurations. Please take a look at the cc test and the java test for more information.

Reply:
@jiuqiant I've added all the remaining tests now. Though I happened to run into an issue with the final test for the `classify_async` API. PTAL after uncommenting the code and running the tests again.

Reply:
Can you copy and paste the error log? I can take a quick look.

Reply:
@jiuqiant I've fixed it now.